### PR TITLE
Jonmv/require parent geq compile version

### PR DESCRIPTION
**Pull request by jonmv**
@gjoranv please review.

Only throw if the parent project is a hosted Vespa parent.
Use parentArtifact, which is defined even when parent project cannot be generated. 
Fallback to the version of the plugin itself for what the "vespaversion" is, for those without a vespa parent.

### CODE REVIEW

File: vespa-application-maven-plugin/src/main/java/com/yahoo/container/plugin/mojo/ApplicationMojo.java [hunk 1]
```diff
@@ -56,15 +57,30 @@ public void execute() throws MojoExecutionException {
...
-                                        System.currentTimeMillis());
+        Version compileVersion = Version.from(vespaversion);
+
+        MavenProject current = project;
+        while (current.getParent() != null && current.getParent().getParentArtifact() != null)
+            current = current.getParent();
+
+        boolean hasVespaParent = false;
+        Artifact parentArtifact = current.getParentArtifact();
+        if (parentArtifact != null && parentArtifact.getGroupId().startsWith("com.yahoo.vespa")) {
```
Comment (gjoranv) [inline vespa-application-maven-plugin/src/main/java/com/yahoo/container/plugin/mojo/ApplicationMojo.java:73]: Should we add a future proof `startsWith(ai.vespa.)` as well?

Suggest to add a trailing dot to com.yahoo.vespa anyway.

**Review by gjoranv — APPROVED** [review]

**Comment by gjoranv** [comment]
The failing unit test is probably among the usual suspects 
```
MessageBusDocumentApiTestCase>AbstractDocumentApiTestCase.requireThatAsyncSessionWorks:123 » NullPointer
```

File: vespa-application-maven-plugin/src/main/java/com/yahoo/container/plugin/mojo/ApplicationMojo.java (hunk 1 above)
Comment (jonmv) [reply vespa-application-maven-plugin/src/main/java/com/yahoo/container/plugin/mojo/ApplicationMojo.java]: Good points, fixed.

**Comment by jonmv** [comment]
Definitely is.

**Comment by gjoranv** [comment]
👍

**Pull request merged**